### PR TITLE
Move search indexing to a worker

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Moves code around so the lunr code is only depended on from the worker and the worker doesn't pull in any code it doesn't need.

### CODE REVIEW

File: src/documentation/search/extracts.ts [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+import { Extract } from "./common";
 
 export type Position = [number, number];
 
-const sortByStart = (positions: Position[]): Position[] =>
-  sortBy(positions, (p) => p[0]);
+// Avoid lodash in the worker
+export const sortByStart = (positions: Position[]): Position[] => {
+  const copy = [...positions];
+  copy.sort((a, b) => (a[0] > b[0] ? 1 : a[0] < b[0] ? -1 : 0));
```
Comment (microbit-matt-hillsdon) [inline src/documentation/search/extracts.ts:8]: Fairly sure it started off life as this in your implementation!

File: src/documentation/search/search-hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  if (!value) {
+    throw new Error("Missing provider!");
+  }
+  return value;
+};
+
+const SearchProvider = ({ children }: { children: ReactNode }) => {
+  const { exploreToolkit, referenceToolkit } = useToolkitState();
+  useEffect(() => {
+    // Wait for both, we'll be re-run for each at start-up.
```
Comment (microbit-robert) [inline src/documentation/search/search-hooks.tsx:29]: `we'll` is a typo?

File: src/documentation/search/extracts.test.ts [hunk 3]
```diff
@@ -62,3 +62,19 @@ describe("contextExtracts", () => {
     ]);
   });
 });
+
+describe("sortByStart", () => {
+  it("works", () => {
```
Comment (microbit-robert) [inline src/documentation/search/extracts.test.ts:67]: Should we provide a better name for this test?

Comment (microbit-matt-hillsdon) [reply src/documentation/search/extracts.test.ts]: 9d67c7f

File: src/documentation/search/search-hooks.tsx (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply src/documentation/search/search-hooks.tsx]: I was identifying with the effect... for some reason. Replaced with something clearer in d8a586a92baaac98b8d17256dc75204ab16c5e28.

**Review by microbit-robert — APPROVED** [review]
LGTM

**Comment by microbit-matt-hillsdon** [comment]
Merged main and fixed conflicts.

**Pull request merged**